hail/python/hailtop/batch/__init__.py

```diff
@@ -4,14 +4,18 @@
 from .batch_pool_executor import BatchPoolExecutor
 from .backend import LocalBackend, ServiceBackend, Backend
 from .utils import BatchException
+from .functions import _combine, concatenate, plink_merge
```

Comment:
I think I prefer your original intuition to make these `utils`. Functions doesn't feel like a very informative name to our users. It wouldn't matter too much except that we use the module name as the section name in the Docs.

Utilities seems good. Combinators is good, but math-y. Tools?